src/pudl/transform/ferc1.py

```diff
@@ -355,6 +362,71 @@ def align_row_numbers_dbf(
+    additional_categories: list[str] = []
+    expected_categories_to_drop: int = 0
+
+
+def select_dbf_rows_from_category(
+    processed_dbf: pd.DataFrame,
+    processed_xbrl: pd.DataFrame,
+    params: SelectDbfRowsFromCategory,
+) -> pd.DataFrame:
+    """Select DBF rows with values listed or found in XBRL in a categorical-like column.
+
+    The XBRL data often breaks out sub-sections of DBF tables into their own table.
+    These breakout tables are often messy, unstructured portions of a particular
+    schedule or page on the FERC1 PDF. We often want to preserve some of the ways the
+    XBRL data is segmented so we need to be able to select only portions of the DBF
+    table to be concatenated with the XBRL data.
+
+    In mapping DBF data to XBRL data for the tables that rely on their ``row_number``
+    we map each row to its corresponding ``xbrl_factoid``. The standard use of this
+    transformer is to use the ``column_name`` that corresponds to the ``xbrl_factoid``
+    that was merged into the DBF data via :func:`align_row_numbers_dbf` and was
+    converted into a column in the XBRL data via :func:`wide_to_tidy`.
+
+    Note: Often, the unstructured portion of the DBF table that (possibly) sums up into
+    a single value in structured data has the same ``xbrl_factoid`` name in the XBRL
+    tables. By convention, we are employing a pattern in the ``dbf_to_xbrl.csv`` map
+    that involves adding an ``_unstructed`` suffix to the rows that correspond to the
+    unstructured portion of the table. This enables a simple selection of the structured
+    part of the table. When processing the unstructured table, you can either rename the
+    XBRL data's factoid name to include an ``_unstructed`` suffix or you can specify
+    the categories with ``_unstructed`` suffixes using the ``additional_categories``
+    parameter.
+    """
+    # compile the list of categories from the possible options.
+    categories_to_select = []
+    if params.select_based_on_xbrl_category:
+        categories_to_select = categories_to_select + list(
+            processed_xbrl[params.column_name].unique()
+        )
+    if params.additional_categories:
+        categories_to_select = categories_to_select + params.additional_categories
+
+    # check if we are getting the same number of expected categories to drop
+    categories_to_drop = [
+        cat
+        for cat in processed_dbf[params.column_name].unique()
+        if cat not in categories_to_select
+    ]
+    if len(categories_to_drop) != params.expected_categories_to_drop:
+        logger.warning(
+            f"Dropping {len(categories_to_drop)} DBF categories that contain the "
+            f"following values in {params.column_name} but expected "
+            f"{params.expected_categories_to_drop}:"
+            f"{categories_to_drop}"
+        )
+    return processed_dbf[
+        processed_dbf[params.column_name].isin(categories_to_select)
+    ].copy()
```

Comment:
I'm having a hard time understanding what is being selected from the dataframe here. Can this be broken down into simpler parts? Or done with a `df.loc[]` maybe?

Reply:
I could break out the mask here. Does that make it more clear?

```
    # now select only the rows which contain the categories we want to include in the
    # column that we care about. Copy bc this is a slice of the og dataframe.
    category_mask = processed_dbf[params.column_name].isin(categories_to_select)
    return processed_dbf.loc[:, category_mask].copy()
```

Otherwise this is what it is with just a loc... it's basically the same thing:
  ```
   # now select only the rows which contain the categories we want to include in the
    # column that we care about. Copy bc this is a slice of the og dataframe.
    return processed_dbf.loc[
        :, processed_dbf[params.column_name].isin(categories_to_select)
    ].copy()
```

Reply:
i went with the first option here. i'm not sure it is more clear so lmk if this does help